WalletWasabi/BitcoinCore/Mempool/MempoolMirror.cs

```diff
@@ -116,24 +116,19 @@ private async Task<int> MirrorMempoolAsync(CancellationToken cancel)
 		return added;
 	}
 
-	public IEnumerable<Transaction> GetSpenderTransactions(IEnumerable<OutPoint> txOuts)
+	public IReadOnlySet<Transaction> GetSpenderTransactions(IEnumerable<OutPoint> txOuts)
 	{
-		Dictionary<uint256, Transaction> spenders = new();
 		lock (MempoolLock)
 		{
-			foreach (var input in txOuts)
-			{
-				foreach (var mempoolTx in Mempool)
-				{
-					if (mempoolTx.Value.Inputs.Select(x => x.PrevOut).Contains(input))
-					{
-						spenders.Add(mempoolTx.Key, mempoolTx.Value);
-					}
-				}
-			}
+			var mempoolTxs = Mempool.Values;
+			var txOutsSet = txOuts.ToHashSet();
+
+			return mempoolTxs.SelectMany(tx => tx.Inputs.Select(i => (tx, i.PrevOut)))
+				.Where(x => txOutsSet.Contains(x.PrevOut))
+				.Select(x => x.tx)
+				.Distinct()
+				.ToHashSet();
```

Comment:
Not reviewing this PR, just noticed distinct is redundant if you tohashset

```suggestion
				.Select(x => x.tx)
				.ToHashSet();
```